packages/workshop-app/start.js

```diff
@@ -1,16 +1,18 @@
-const dotenv = require('dotenv')
-const path = require('path')
+import dotenv from 'dotenv'
+import path from 'path'
 
 dotenv.config({
 	path: path.join(process.env.KCDSHOP_CONTEXT_CWD ?? process.cwd(), '.env'),
 })
 
 if (process.env.NODE_ENV === 'production') {
-	require('./build/server')
+	await import('./build/server/index.js')
 } else {
+	// FIXME: global.__inspector_open__ is always undefined here, even when the inspectore runing
```

Comment:
I don't understand how that could be possible... I do want to add this to the Epic Stack as well.

Reply:
maybe `tsx` starts a new process on every file change?

I've checked the process.pid and it change every time the file changed

Reply:
@kentcdodds, how about adding `--inspect` to the end of the npm script ?

```diff
- "start:server": "tsx watch --clear-screen=false --ignore \"app/**\" --ignore \"build/**\" --ignore \"utils/**\" --ignore \"node_modules/**\" ./start.js",
+ "start:server": "tsx watch --clear-screen=false --ignore \"app/**\" --ignore \"build/**\" --ignore \"utils/**\" --ignore \"node_modules/**\" --inspect ./start.js",
```

Reply:
'global.__server_close_with_grace_return__' is also always undefined in `server/index.js`

Reply:
Oh, I think the reason for this is because the `start.js` file is never re-required. It's only ever run once. So the global and the if statement are probably unnecessary.

Reply:
but what about 'global.server_close_with_grace_return' it is on server/index.js that we do import,
as i mention about, it look to me that tsx generate new process on every change.

regarding   inspector we can simply start it from the npm command.   `"start:server": "tsx watch ..... --inspect ./start.js"`

Reply:
> global.server_close_with_grace_return

Yes, that one can also be removed.

I'm fine switching to the command line flag

Reply:
done in df748ed